grpc/tests/src/test/system_costs.rs

```diff
@@ -0,0 +1,927 @@
+        let rhs = balance_before - call_cost;
+        assert_eq!(
+            lhs,
+            rhs,
+            "Calling a failed entrypoint {} does not incur expected cost of {} but {}",
+            entrypoint,
+            expected_cost,
+            if lhs > rhs { lhs - rhs } else { rhs - lhs }
+        );
+        assert_eq!(builder.last_exec_gas_cost().value(), call_cost);
+    }
+}
+
+#[ignore]
+#[test]
+fn should_not_charge_system_account_for_running_auction() {
+    // This verifies a corner case where system account calls `run_auction` entrypoint, but it
+    // shouldn't be charged for doing so. Otherwise if that happens the system could fail as in
+    // production code system account is most likely empty.
+    let fund_system_account_request = ExecuteRequestBuilder::standard(
+        *DEFAULT_ACCOUNT_ADDR,
+        CONTRACT_TRANSFER_TO_ACCOUNT,
+        runtime_args! { ARG_TARGET => SYSTEM_ACCOUNT_ADDR, ARG_AMOUNT => U512::from(MINIMUM_ACCOUNT_CREATION_BALANCE) },
+    )
+    .build();
+
+    let mut builder = InMemoryWasmTestBuilder::default();
+
+    builder.run_genesis(&*DEFAULT_RUN_GENESIS_REQUEST);
+
+    let auction_hash = builder.get_auction_contract_hash();
+
+    builder
+        .exec(fund_system_account_request)
+        .commit()
+        .expect_success();
+
+    let system_account = builder
+        .get_account(SYSTEM_ACCOUNT_ADDR)
+        .expect("should have system account");
+    let default_account = builder
+        .get_account(*DEFAULT_ACCOUNT_ADDR)
+        .expect("should have default account");
+
+    let run_auction_as_system_request = ExecuteRequestBuilder::contract_call_by_hash(
+        SYSTEM_ACCOUNT_ADDR,
+        auction_hash,
+        auction::METHOD_RUN_AUCTION,
+        RuntimeArgs::default(),
+    )
+    .build();
+
+    let run_auction_as_user_request = ExecuteRequestBuilder::contract_call_by_hash(
+        *DEFAULT_ACCOUNT_ADDR,
+        auction_hash,
+        auction::METHOD_RUN_AUCTION,
+        RuntimeArgs::default(),
+    )
+    .build();
+
+    // Verify that system account is not charged
+    let system_funds_before = builder.get_purse_balance(system_account.main_purse());
+    builder
+        .exec(run_auction_as_system_request)
+        .commit()
+        .expect_success();
+    let system_funds_after = builder.get_purse_balance(system_account.main_purse());
+
+    assert_eq!(system_funds_before, system_funds_after);
+
+    let user_funds_before = builder.get_purse_balance(default_account.main_purse());
+    builder.exec(run_auction_as_user_request).commit();
+    let user_funds_after = builder.get_purse_balance(default_account.main_purse());
+
+    let expected_call_cost = U512::from(DEFAULT_PAY_COST) + U512::from(DEFAULT_RUN_AUCTION_COST);
+    let lhs = user_funds_after;
+    let rhs = user_funds_before - expected_call_cost;
+    assert_eq!(
+        lhs,
+        rhs,
+        "unexpected difference {}",
+        if lhs > rhs { lhs - rhs } else { rhs - lhs },
```

Comment:
Could we remove these?  They may have been helpful for you while developing, but it's much easier for me to just read the normal output of this macro when it fails.

(and below)

Reply:
Done